src/transformers/pipelines/audio_classification.py

```diff
@@ -110,12 +110,18 @@ def __call__(
         information.
 
         Args:
-            inputs (`np.ndarray` or `bytes` or `str`):
-                The inputs is either a raw waveform (`np.ndarray` of shape (n, ) of type `np.float32` or `np.float64`)
-                at the correct sampling rate (no further check will be done) or a `str` that is the filename of the
-                audio file, the file will be read at the correct sampling rate to get the waveform using *ffmpeg*. This
-                requires *ffmpeg* to be installed on the system. If *inputs* is `bytes` it is supposed to be the
-                content of an audio file and is interpreted by *ffmpeg* in the same way.
+            inputs (`np.ndarray` or `bytes` or `str` or `dict`):
+                The inputs is either :
+                    - `str` that is the filename of the audio file, the file will be read at the correct sampling rate
+                      to get the waveform using *ffmpeg*. This requires *ffmpeg* to be installed on the system.
+                    - `bytes` it is supposed to be the content of an audio file and is interpreted by *ffmpeg* in the
+                      same way.
+                    - (`np.ndarray` of shape (n, ) of type `np.float32` or `np.float64`)
+                        Raw audio at the correct sampling rate (no further check will be done)
+                    - `dict` form can be used to pass raw audio sampled at arbitrary `sampling_rate` and let this
+                      pipeline do the resampling. The dict must be either be in the format `{"sampling_rate": int,
+                      "raw": np.array}`, or `{"sampling_rate": int, "array": np.array}`, where the key `"raw"` or
+                      `"array"` is used to denote the raw audio waveform.
```

Comment:
Copied one-for-one from: https://github.com/huggingface/transformers/blob/5dfd407b37ac683dc91637e9913b0ae9205d2acd/src/transformers/pipelines/automatic_speech_recognition.py#L235

Reply:
Saw the raw key on the ASR pipeline was kept for backward compatibility. Do we really need to introduce it there?

Reply:
Let's see what @Narsil thinks.

Reply:
Gently pinging @Narsil - would be nice to have this ready in `transformers` for the next release (unblocks https://github.com/huggingface/audio-transformers-course/pull/25)

Reply:
Hey, sorry for this.

Usually I'm kind of against accepting widely different types, this case is difference since it's about our ecosystem and making `datasets` + `pipeline` work together nicer.

Reply:
If we're all happy I'll keep it as is then for now and we can explore a joint refactor of the ASR + audio class pipelines in the future?

Reply:
Yup